### PR TITLE
Naturalness Calculation

### PR DESCRIPTION
**Pull request by elmiele**
Calculate naturalness score for a given protein-CDS pair using GEMORNA model. slightly higher than the Supp. Data(0.01-0.03)

### CODE REVIEW

File: calculate_naturalness.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+        enc_prot = model.encoder(prot_tensor, prot_mask)
+        output, _ = model.decoder(cds_input, enc_prot, cds_mask, prot_mask)
+
+        # Get probabilities
+        log_probs = F.log_softmax(output, dim=-1)
+        target_log_probs = log_probs.gather(2, cds_target.unsqueeze(-1)).squeeze(-1).squeeze(0)
+        probs = torch.exp(target_log_probs)
+
+        # Compute metrics
+        naturalness = probs.mean().item()
```
Comment (GoodCoder666) [inline calculate_naturalness.py]: The naturalness score is mathematically the **Geometric Mean** of probabilities, not the Arithmetic Mean.
> Naturalness score $N_s$ is the exponential form of the length-normalized model score.
```suggestion
        naturalness = torch.exp(target_log_probs.mean()).item()
```
Or simply use `1 / perplexity`.

**Review by GoodCoder666 — CHANGES_REQUESTED** [review]

File: calculate_naturalness.py (hunk 1 above)
Comment (GoodCoder666) [reply calculate_naturalness.py]: This probably explains why your results are higher than expected (AM-GM inequality).
